current/runtime/storage.md

```diff
@@ -4,49 +4,240 @@ lang: en
+non-cryptographic counterparts, which is why Substrate allows developers to select when they are
+used.
+
+##### Transparent Hashing Algorithms
+
+A transparent hashing algorithm is one that makes it easy to discover and verify the input that was
+used to generate a given output. In Substrate, hashing algorithms are made transparent by
+concatenating the algorithm's input to its output. This makes it trivial for users to retrieve a
+key's original unhashed value and verify it if they'd like (by re-hashing it). The creators of
+Substrate have **deprecated the use of non-transparent hashers** within FRAME-based runtimes, so
+this information is provided primarily for completeness. In fact, it is _necessary_ to use a
+transparent hashing algorithm if you would like access [iterable map](#Iterable-Storage-Maps)
+capabilities. Refer to [the advanced storage documentation](../advanced/storage#Storage-Map-Keys) to
+learn more about the important capabilities that transparent hashing algorithms expose.
+
+##### Common Substrate Hashers
+
+This table lists some common hashers used in Substrate and denotes those that are cryptographic and
+those that are transparent:
+
+| Hasher                                                                                     | Cryptographic | Transparent |
+| ------------------------------------------------------------------------------------------ | ------------- | ----------- |
+| [Blake2 128 Concat](https://crates.parity.io/frame_support/struct.Blake2_128Concat.html)   | X             | X           |
+| [TwoX 64 Concat](https://crates.parity.io/frame_support/struct.Twox64Concat.html)          |               | X           |
+| [Identity](https://crates.parity.io/frame_support/struct.Identity.html)                    |               |             |
+| [Blake2 128](https://crates.parity.io/frame_support/struct.Blake2_128.html) **DEPRECATED** | X             |             |
+| [TwoX 128](https://crates.parity.io/frame_support/struct.Twox128.html) **DEPRECATED**      |               |             |
+
+The Identity hasher encapsulates a hashing algorithm that has an output equal to its input (the
+identity function). This type of hasher should only be used when the starting key is already a
+cryptographic hash.
+
+## Declaring Storage Items
+
+You can use
+[the `decl_storage` macro](https://crates.parity.io/frame_support/macro.decl_storage.html) to easily
+create new runtime storage items. Here is an example of what it looks like to declare each type of
+storage item:
 
 ```rust
 decl_storage! {
 	trait Store for Module<T: Trait> as Example {
-		pub SomeValue: u64;
-		pub SomeMap: map u64 => u64;
-		pub SomeLinkedMap: linked_map u64 => u64;
-		pub SomeDoubleMap: double_map u64, blake2_256(u64) => u64;
+		SomePrivateValue: u32;
+		pub SomePrimitiveValue get(fn some_primitive_value): u32;
+		// complex types are prefaced by T::
+		pub SomeComplexValue: T::AccountId;
```

Comment:
I feel like the wording shows that there is macro magic in the syntax here. but actually this is a regular rust type.
(The macro parse the type themself to know if there are generic over `T`, if they are then the storage struct generated is generic as well.)

```suggestion
		// types can make use of generic `T: Trait`
		pub SomeComplexValue: T::AccountId;
```